src/arg.c

```diff
@@ -110,6 +110,111 @@ static void dcc_note_compiled(const char *input_file, const char *output_file)
            "compile from %s to %s", input_base, output_base);
 }
 
+/* Look for march/mtune=native in the arguments, and replace with the
+ * actual arguments employed by gcc / clang. */
+static int dcc_resolve_march_native(char* argv[], char*** ret_newargv, int extra_args, int* ignore_range_min, int* ignore_range_max)
+{
+    int i, l, j;
+    l = dcc_argv_len(argv);
+    if (l == 0) return 0;
+    const char* compiler = strrchr(argv[0], '/');
```

Comment:
This is going to require switching to c11 mode (which is default in all compilers except the ancient one that travisCI uses), but that is OK.